### PR TITLE
scala-starter-bot

### PR DESCRIPTION
**Pull request by niekz**
Added a starter bot for the scala language.

Added dockerfile, readme, and source for the bot.

Dockerfile needs clarification, as the example dockerfile provides little to no detail on usage.

### CODE REVIEW

**Comment by Privolin** [comment]
Checked Docker Container:
Build Container - Successful ✔️
Build Sample Bot - Successful ✔️
Run Starter Bot (independently) - Successful ✔️

File: starter-bots/scala/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM entelectchallenge/base:2019
+
+RUN apt-get update -y
+
+RUN apt-get install default-jre -y
+RUN apt-get install default-jdk -y
```
Comment (DewaldDeJager) [inline starter-bots/scala/Dockerfile]: It should be possible to omit the first line as the JRE is bundled with the JDK.

File: starter-bots/scala/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Java starter bot
```
Comment (DewaldDeJager) [inline starter-bots/scala/README.md]: This should be Scala 🙂

File: starter-bots/scala/src/main/scala/za/co/entelect/challenge/Bot.scala [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    else if(horizontalComponent > 0) builder.append('E')
+
+    Directions.withNameInsensitive(builder.toString())
+  }
+
+  private def getSurroundingCells(x: Int, y: Int): List[Cell] = {
+    val cells: ListBuffer[Cell] = ListBuffer.empty[Cell]
+
+    for (i <- Range(x - 1, x + 2)) {
+      for (j <- Range(y - 1, y + 2)) {
```
Comment (DewaldDeJager) [inline starter-bots/scala/src/main/scala/za/co/entelect/challenge/Bot.scala]: Consider putting both of these into one `for` statement to increase readability, eg:
```scala
for (
  i <- Range(x - 1, x + 2);
  j <- Range(y - 1, y + 2)
) {
```

File: starter-bots/scala/src/main/scala/za/co/entelect/challenge/Main.scala [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import scala.util.Random
+import org.json4s._
+import org.json4s.native.JsonMethods._
+import za.co.entelect.challenge.commands.Command
+import za.co.entelect.challenge.entities.GameState
+import za.co.entelect.challenge.enums.Serializers
+
+object Main {
+  private final val ROUNDS_DIRECTORY = "rounds"
+  private final val STATE_FILE_NAME  = "state.json"
```
Comment (DewaldDeJager) [inline starter-bots/scala/src/main/scala/za/co/entelect/challenge/Main.scala]: `final` is not necessary here. The reason for this is that `final` prevents the fields from being overridden. These fields are both private and in a singleton object, so they cannot be overridden 🙂

**Comment by niekz** [comment]
@DewaldDeJager Updated based on the comments above :)

**Pull request merged**